### DISCUSSION TITLE
Can't install swig3.0 in docker container using airflow2.3.2-python3.8

### DISCUSSION THREAD

**Discussion by HiGal** [apache/airflow#25308] [General]
### Apache Airflow version

2.3.2

### What happened

I am trying to install swig3.0 inside airflow container for Jamspell using following Dockerfile
```Dockerfile
FROM apache/airflow:2.3.2-python3.8

USER root

RUN apt-get update \
  && apt-get install -y --no-install-recommends \
         build-essential git \
  && apt-get autoremove -yqq --purge \
  && apt-get clean \
  && rm -rf /var/lib/apt/lists/* \
  && apt-get update \
  && apt install -y swig3.0
```

I got following error
```
Reading package lists...
Building dependency tree...
Reading state information...
E: Unable to locate package swig3.0
E: Couldn't find any package by glob 'swig3.0'
```

### What you think should happen instead

When I used previous version 2.1.0 I didn't meet those error.

### How to reproduce

To reproduce you should try to build following docker image from this Dockerfile:
```Dockerfile
FROM apache/airflow:2.3.2-python3.8

USER root

RUN apt-get update \
  && apt-get install -y --no-install-recommends \
         build-essential git \
  && apt-get autoremove -yqq --purge \
  && apt-get clean \
  && rm -rf /var/lib/apt/lists/* \
  && apt-get update \
  && apt install -y swig3.0
```

### Operating System

MacOS

### Deployment

Docker-Compose

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by boring-cyborg** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template!

**Reply by potiuk** [reply]
This is not a problem of airflow (and certainly not a bug). Converted into a discussion.

You need to check how to install swig on debian bullseye and do it the same. Airflow image is based on debian bulleseye and whatever you  need to install it on bullseye will work for airflow. Airlfow used to use "buster" version in 2.1.0 but we switched to bullseye.

You can either find out how to install swig (maybe different version?) on Bullseye or attempt to build an image following "customization" route https://airflow.apache.org/docs/docker-stack/build.html#choosing-debian-version-when-customizing-the-image choosing buster as an image (we are generally supporting Buster https://airflow.apache.org/docs/docker-stack/changelog.html#airflow-2-3 until the EOL). 

Not however that Buster's EOL is August 2022 so in about a month we will stop supporting buster even if you want to build custom image, so I advice you to find a way to install swig on Bullseye.

**Reply by potiuk** [reply]
For the reference - see discussion from Feb when we discussed when and how we swtich to Debian Bullseye: https://lists.apache.org/thread/fo20nqb8gs449os1vogjqdd1rv0pxx79

**Reply by potiuk** [reply]
And BTW. Bullseye support Swig4 - so probably that's your best choice: https://packages.debian.org/bullseye/swig

**Reply by HiGal** [reply]
I am running such command
```
docker build . -t <image_name> --build-arg PYTHON_BASE_IMAGE=python:3.8-buster
```
but receiving the same assertion error

**Reply by potiuk** [reply]
You should build the image following customisation route (i.e Build airrlow image from the scratch) not following the "extension" route. You shoudl read the docs carefuly (there are examples and steps to follow) and you can watch my talk explaining the difference https://www.youtube.com/watch?v=wDr3Y7q2XoI
